services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/liitteet/AttachmentsController.kt

```diff
@@ -0,0 +1,131 @@
+    @Autowired private val hankeService: HankeService,
+    @Autowired private val permissionService: PermissionService,
+) {
+
+    @GetMapping("/{uuid}")
+    @Operation(
+        summary = "Get attachment metadata by UUID",
+        description =
+            """
+                Return information about an attachment. Available even when scan is pending.
+            """
+    )
+    fun getAttachment(@PathVariable uuid: UUID): AttachmentMetadata {
+        val currentUserId = currentUserId()
+        val info = attachmentService.get(uuid)
+        val hankeId = hankeService.getHankeId(info.hankeTunnus)
+
+        if (
+            hankeId == null ||
+                !permissionService.hasPermission(hankeId, currentUserId, PermissionCode.VIEW)
+        ) {
+            throw HankeNotFoundException("")
+        }
+
+        return info
+    }
+
+    @GetMapping("/{uuid}/content")
+    @Operation(
+        summary = "Download attachment file.",
+        description =
+            """
+                Returns the uploaded file. Only available after succesful virus scan.
+            """
+    )
+    fun getAttachmentContent(
+        @PathVariable uuid: UUID,
+        response: HttpServletResponse
+    ): HttpEntity<ByteArray> {
+        val currentUserId = currentUserId()
+        val info = attachmentService.get(uuid)
+        val hankeId = hankeService.getHankeId(info.hankeTunnus)
+
+        if (
+            hankeId == null ||
+                !permissionService.hasPermission(hankeId, currentUserId, PermissionCode.VIEW)
+        ) {
+            throw HankeNotFoundException("")
+        }
+
+        val metadata = attachmentService.get(uuid)
+        val mimeType = URLConnection.guessContentTypeFromName(metadata.name)
+
+        val headers = HttpHeaders()
+        headers.contentType = MediaType.parseMediaType(mimeType)
+        response.setHeader(HttpHeaders.CONTENT_DISPOSITION, "attachment; filename=" + metadata.name)
+
+        val file = attachmentService.getContent(uuid)
+        return HttpEntity<ByteArray>(file, headers)
+    }
+
+    @DeleteMapping("/{uuid}")
+    @Operation(
+        summary = "Remove an attachment",
+        description =
+            """
+                Removes an attachment and the related connections.
+            """
+    )
+    fun removeAttachment(@PathVariable uuid: UUID) {
+        val currentUserId = currentUserId()
+        val info = attachmentService.get(uuid)
+        val hankeId = hankeService.getHankeId(info.hankeTunnus)
+
+        if (
+            hankeId == null ||
+                !permissionService.hasPermission(hankeId, currentUserId, PermissionCode.EDIT)
+        ) {
+            throw HankeNotFoundException("")
+        }
+
+        return attachmentService.removeAttachment(uuid)
+    }
+
+    @ResponseStatus(HttpStatus.NOT_FOUND)
+    @ExceptionHandler(HankeNotFoundException::class)
+    fun handleArgumentExceptions(ex: HankeNotFoundException): HankeError {
```

Comment:
`HankeNotFoundException` is already handled in `ControllerExceptionHandler`.

Reply:
This is removed.